components/function-controller/design/glossary_proposal.md

```diff
@@ -0,0 +1,105 @@
+# Serverless naming convention
+
+## Problem overview
+
+Currently, Serverless in Kyma consists of two projects:
+
+- [function-controller](https://github.com/kyma-project/kyma/tree/main/components/function-controller) -
+  responsible for running a Function on a k8s cluster
+- [serverless-manager](https://github.com/kyma-project/serverless-manager) - responsible for installation and
+  configuration of Serverless
+
+Additionally, we have 3rd-party additional components like [KEDA](https://keda.sh/)
+
+In Serverless, we overuse the word "controller" which causes confusion and requires clarification. Saying "controller", we refer to:
+
+- Serverless reconcile loop
+- Serverless Pod with the reconcile loop
+- the Function Controller component in the `kyma` directory
+
+## Goal
+
+The goal of this document is to clarify the naming convention in Serverless and define its elements to avoid confusion and make it more logical.
+
+## Proposal
+
+The proposed naming conventions refer to different architecture layers of the whole project. See the [architecture](./assets/kubebuilder-architecture.png) diagram for details.
+
+### Project naming convention
+
+This section refers to the high-level architecture elements, namely to the main projects:
+
+- Serverless - the new naming convention for the `function-controller`. Serverless is responsible for running a Function on a k8s cluster. It can contain its own
```

Comment:
```suggestion
- Serverless - the new naming convention for the `function-controller`. Serverless is responsible for running a Function on a Kubernetes cluster. It can contain its own
```